lib/ansible/modules/cloud/amazon/ec2_vpc_subnet.py

```diff
@@ -102,64 +237,115 @@ def get_subnet_info(subnet):
         subnet = camel_dict_to_snake_dict(subnet)
 
     if 'tags' in subnet:
-        subnet['tags'] = boto3_tag_list_to_ansible_dict(subnet['tags'])
+        subnet['tags'] = boto3_tag_list_to_ansible_dict(subnet['tags'], 'key', 'value')
     else:
-        subnet['tags'] = dict()
+        subnet['tags'] = [dict()]
 
     if 'subnet_id' in subnet:
         subnet['id'] = subnet['subnet_id']
         del subnet['subnet_id']
 
+    subnet['ipv6_cidr_block'] = ''
+    subnet['ipv6_association_id'] = ''
+    ipv6set = subnet.get('ipv6_cidr_block_association_set')
+    if ipv6set:
+        for item in ipv6set:
+            if item.get('ipv6_cidr_block_state', {}).get('state') in ('associated', 'associating'):
+                subnet['ipv6_cidr_block'] = item['ipv6_cidr_block']
+                subnet['ipv6_association_id'] = item['association_id']
+
     return subnet
 
 
 def subnet_exists(conn, subnet_id):
     filters = ansible_dict_to_boto3_filter_list({'subnet-id': subnet_id})
     subnets = get_subnet_info(conn.describe_subnets(Filters=filters))
-    if len(subnets) > 0 and 'state' in subnets[0] and subnets[0]['state'] == "available":
+    if subnets and 'state' in subnets[0] and subnets[0]['state'] == "available":
         return subnets[0]
-    else:
-        return False
 
+    return False
+
+
+def create_subnet(conn, module, **kwargs):
```

Comment:
Moving to `**kwargs` makes this function more opaque, since by looking you can't tell what's used and what's not. Can you put back the mandatory kwargs and make any optional ones defaulted?